freelist.go

```diff
@@ -6,18 +6,29 @@ import (
 	"unsafe"
 )
 
+
+// txPending holds a list of pgids and corresponding allocation txns
+// that are pending to be freed.
+type txPending struct {
+	ids              []pgid
+	alloctx          []txid // txids allocating the ids
```

Comment:
why do we need to record alloctx. i thought the pages will only be freed/allocated by RW tx. So when we decide to free it up, the allocator tx should be closed (allocated by a previous rw tx) already or closing (allocated by this tx).